schemas/v3.1/schema.yaml

```diff
@@ -0,0 +1,911 @@
+            type: boolean
+        allOf:
+          - $ref: '#/$defs/examples'
+          - $ref: '#/$defs/parameter/dependentSchemas/schema/$defs/styles-for-path'
+          - $ref: '#/$defs/parameter/dependentSchemas/schema/$defs/styles-for-header'
+          - $ref: '#/$defs/parameter/dependentSchemas/schema/$defs/styles-for-query'
+          - $ref: '#/$defs/parameter/dependentSchemas/schema/$defs/styles-for-cookie'
+          - $ref: '#/$defs/parameter/dependentSchemas/schema/$defs/styles-for-form'
+
+        $defs:
+          styles-for-path:
+            if:
+              properties:
+                in:
+                  const: path
+              required:
+                - in
+            then:
+              properties:
+                style:
+                  default: simple
+                  enum:
+                    - matrix
+                    - label
+                    - simple
+                required:
+                  const: true
+              required:
+                - required
+
+          styles-for-header:
+            if:
+              properties:
+                in:
+                  const: header
+              required:
+                - in
+            then:
+              properties:
+                style:
+                  default: simple
+                  enum:
+                    - simple
+
+          styles-for-query:
+            if:
+              properties:
+                in:
+                  const: query
+              required:
+                - in
+            then:
+              properties:
+                style:
+                  default: form
+                  enum:
+                    - form
+                    - spaceDelimited
+                    - pipeDelimited
+                    - deepObject
+
+          styles-for-cookie:
+            if:
+              properties:
+                in:
+                  const: cookie
+              required:
+                - in
+            then:
+              properties:
+                style:
+                  default: form
+                  enum:
+                    - form
+
+          styles-for-form:
+            if:
+              properties:
+                style:
+                  const: form
+              required:
+                - style
+            then:
+              properties:
+                explode:
+                  default: true
+            else:
+              properties:
+                explode:
+                  default: false
+
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+
+  parameter-or-reference:
+    if:
+      required:
+        - $ref
+    then:
+      $ref: '#/$defs/reference'
+    else:
+      $ref: '#/$defs/parameter'
+
+  request-body:
+    type: object
+    properties:
+      description:
+        type: string
+      content:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/media-type'
```

Comment:
The property names of this object should be a media type range. We can change this to `patternProperties` and add a regex that enforces that, but such a regex would be pretty ugly. Not sure it's worth it. Maybe something simple but incomplete is a reasonable compromise.

Reply:
I'm sure this could be cleaned up a bit, but I don't think a strict regex option can do much better than this monstrosity. 
```
/(?:\*\/\*|(?:[!#$%&'+\-.^`|~]|\w)+\/\*|(?:[!#$%&'+\-.^`|~]|\w)+\/(?:[!#$%&'+\-.^`|~]|\w)+)(?:\s*;\s*(?:[!#$%&'*+\-.^`|~]|\w)+=(?:(?:[!#$%&'*+\-.^`|~]|\w)+|"(?:[\t !\x23-\x5B\x5D-\x7E\x80-\xFF]|\\[\t \x21-\x7E\x80-\xFF])*"))*/
```
Another option is to add this,

```
propertyNames:
  format: media-type-range
```

Of course "media-range-type" isn't a defined value for `format`, but many validators allow plugging in support for custom formats. So, it might be useful in some cases.

Reply:
Here's a somewhat reasonable loose alternative regex. It's a little more restrictive than spec in some places as a little more restrictive in others. 

```
^[\w*]+\/[\w*+-]+(?:\s*;\s*\w+=(?:[^"/]+?|"(\\.|[^"])*"))*$
```

Reply:
As per the TSC conversation, let's use the media-type-range.